test/jit/test_cuda.py

```diff
@@ -49,6 +49,35 @@ def tearDown(self):
         torch.cuda.empty_cache()
         super(TestCUDA, self).tearDown()
 
+    @skipIfRocm
+    @unittest.skipIf(not TEST_MULTIGPU, "detected only one GPU")
+    def test_cuda_synchronize(self):
+        # Test device synchronization.
+
+        @torch.jit.script
+        def test_device_synchronize():
+            prev_current_device_index = torch.cuda.current_device()
+            torch.cuda.synchronize()
+            torch.cuda.synchronize('cuda')
+            torch.cuda.synchronize('cuda:0')
+            torch.cuda.synchronize(0)
+            torch.cuda.synchronize(torch.device('cuda:1'))
+            after_current_device_index = torch.cuda.current_device()
+
+            return prev_current_device_index == after_current_device_index
+
+        @torch.jit.script
+        def test_multi_device_synchronize():
+            torch.cuda.synchronize(torch.device('cuda:0'))
+            prev_current_device_index = torch.cuda.current_device()
+            torch.cuda.synchronize(1)
+            after_current_device_index = torch.cuda.current_device()
+
+            return prev_current_device_index == after_current_device_index
```

Comment:
Consider adding a comment explaining why this check is needed.

Reply:
Done